### PR TITLE
added  'ja' translation for some article

### PR DESCRIPTION
**Pull request by hand-dot**
Hello Dan!

I was added A Japanese translation  to the following blog.
-  Why do we write super (props)?
-  How does React distinguish classes from functions?
-  Why does the reactive element have a $$ type property?
-  How does setState know what to do?

and added language code `ja` in `i18n.js`.

Thanks always for your interesting blog! 🎉

### CODE REVIEW

**Comment by gaearon** [comment]
So cool! Thanks

File: src/pages/how-does-react-tell-a-class-from-a-function.ja.md [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+title: Reactはどうやって関数からクラスを見分けるているの？
+date: '2018-12-02'
+langs: ['en', 'ja']
+spoiler: クラス、new、instanceof、プロトタイプチェーン、およびAPI設計について説明します。
+---
+
+関数として定義された`Greeting`コンポーネントについて考えてみましょう:
+
+```jsx
+function Greeting() {
```
Comment (j-f1) [inline src/pages/how-does-react-tell-a-class-from-a-function.ja.md]: You can translate these, too if you’d like — A quick test shows that Japanese characters are valid function names.

Comment (j-f1) [reply src/pages/how-does-react-tell-a-class-from-a-function.ja.md]: Not sure what the conventions are for writing code in languages other than English, though.

Comment (gaearon) [reply src/pages/how-does-react-tell-a-class-from-a-function.ja.md]: I think for consistency I'd leave code snippets original (except comments)

**Pull request merged**